### ISSUE TITLE
"error: Expected { on line" with a guard and a double subscript

### ISSUE THREAD

**Issue by juri** [nicklockwood/SwiftFormat#1095]
This is a valid Swift file:

```swift
#!/usr/bin/swift

let array: [[String: Any]] = [
    ["foo": "bar"],
]

guard let value = array[0]["foo"] as? String else {
    fatalError("oops")
}

print("found value:", value)
```

Running SwiftFormat 0.49.1 on it with no configuration reports the following:

```
swiftformat --lint Guard.swift
Running SwiftFormat...
(lint mode - no files will be changed.)
warning: No Swift version was specified, so some formatting features were disabled. Specify the version of Swift you are using with the --swiftversion option, or by adding a .swift-version file to your project.
error: Expected { on line 7 in /path/to/Guard.swift.
```

It looks like it requires that double subscript to trigger it, it didn't happen with a simple dict.

This worked with SwiftFormat 0.48.x, but seems to have broken with 0.49.

**Comment by zacwest** [comment]
This also occurs for a function returning a dictionary, e.g.:

```swift
if let test = value()["hi"] {
    print("hi")
}
```

produces "error: Expected { on line 1"

**Comment by nicklockwood** [comment]
@juri @zacwest fixed in 0.49.2

**Comment by shkhaliq** [comment]
Still running into this on `0.49.2`:

```
guard let row: Row = promotionSections[indexPath.section][indexPath.row] else { return UITableViewCell() }
```

**Comment by nicklockwood** [comment]
@shkhaliq I'm not able to reproduce the error on 0.49.2 with a file containing just that line. Can you provide a standalone example that demonstrates the problem? Or perhaps something in your .swiftformat config is needed to reproduce it?

**Comment by shkhaliq** [comment]
Okay how about you try this in an empty file. I can reproduce it then

```
    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        guard let row: Row = promotionSections[indexPath.section][indexPath.row] else { return UITableViewCell() }
        let cell = tableView.dequeueReusable(RowTableViewCell.self, forIndexPath: indexPath)
        cell.update(row: row)
        return cell
    }
   
  ```

**Comment by nicklockwood** [comment]
@shkhaliq OK, I can reproduce with this example, thanks 👍

**Comment by nicklockwood** [comment]
@shkhaliq this appears to be fixed in 0.49.3

**Comment by shkhaliq** [comment]
@nicklockwood Yes thank you for the fix!
